### ISSUE TITLE
Clean up pool_type logic

### ISSUE THREAD

**Issue by sleepinyourhat** [nyu-mll/jiant#631]
Alex expressed concern that this option may not work.

```
pool_type = max // The type of pooling to aggregate sequences of vectors into a single vector.
                // Options: first, last, max, mean.
```

The easy/temporary fix would be to simply delete the option.

**Comment by sleepinyourhat** [comment]
Not resolved yet, but bumping to needed-for-1.0, since it's no longer a major issue.

**Comment by sleepinyourhat** [comment]
Still on it? This is the last day I can do much reviewing before Mon.

**Comment by W4ngatang** [comment]
Resolved with #802
